megamek/src/megamek/common/Coords.java

```diff
@@ -350,26 +301,15 @@ public boolean equals(Object object) {
             return false;
         }
         Coords other = (Coords) object;
-        return other.getX() == this.getX() && other.getY() == this.getY();
+        return (other.getX() == x) && (other.getY() == y);
     }
 
-    /* 
-     * Determines if this set of coordinates is on the edge of the board
-     */
-    public boolean isOnBoardEdge(IBoard board) {
-        return (getX() == 0) 
-                || (getY() == 0)
-                || (getX() == (board.getWidth() - 1)) 
-                || (getY() == (board.getHeight() - 1));
-    }
-    
-    /**
-     * Get the hash code for these coords.
-     * 
-     * @return The <code>int</code> hash code for these coords.
-     */
+    /** Returns the hash code for these coords. */
     @Override
     public int hashCode() {
+        if (hash == 0) {
+            hash = (HashCodeUtil.hash1(x + 1337) ^ HashCodeUtil.hash1(y + 97331)) & 0x7FFFFFFF;
```

Comment:
Why not `Objects.hash(x, y)`?

Reply:
Akjosch added HashCodeUtil. It has the comment:
Specialized hash code implementations suitable for PRNG initialisation
I have no idea why. I just left it as it was.

Reply:
Changed it now.